File: apps/Repository/Apps/WebAddress.cs
namespace Allors.Repository.Domain
{
    using System;

    #region Allors
    [Id("5138c0e3-1b28-4297-bf45-697624ee5c19")]
    #endregion
    public partial class WebAddress : ElectronicAddress 
    {
        #region inherited properties
        public string ElectronicAddressString { get; set; }

        public string Description { get; set; }

        public ContactMechanism[] FollowTo { get; set; }

        public Permission[] DeniedPermissions { get; set; }

        public SecurityToken[] SecurityTokens { get; set; }

        #endregion


        #region inherited methods


        public void OnBuild(){}

        public void OnPostBuild(){}

        public void OnPreDerive(){}

        public void OnDerive(){}

        public void OnPostDerive(){}





        public void Delete(){}
        #endregion

    }
}

File: Domains/Apps/Workspace/Typescript/Domain/src/WebAddress.ts
import { WebAddress } from "@allors/generated/dist/domain/WebAddress.g";

declare module "@allors/generated/dist/domain/WebAddress.g" {
  interface WebAddress {
    displayName;
  }
}

Object.defineProperty(WebAddress.prototype, "displayName", {
  get(this: WebAddress) {
    if (this.ElectronicAddressString) {
        return this.ElectronicAddressString;
    }

    return "N/A";
  },
});
